clouddriver-titus/src/main/groovy/com/netflix/spinnaker/clouddriver/titus/deploy/actions/AbstractTitusEnableDisableAction.java

```diff
@@ -0,0 +1,281 @@
+import com.netflix.spinnaker.clouddriver.titus.client.model.TaskState;
+import com.netflix.spinnaker.clouddriver.titus.credentials.NetflixTitusCredentials;
+import com.netflix.spinnaker.clouddriver.titus.deploy.description.EnableDisableInstanceDiscoveryDescription;
+import com.netflix.spinnaker.clouddriver.titus.deploy.description.EnableDisableServerGroupDescription;
+import com.netflix.spinnaker.clouddriver.titus.deploy.ops.discovery.TitusEurekaSupport;
+import com.netflix.spinnaker.kork.exceptions.UserException;
+import com.netflix.titus.grpc.protogen.LoadBalancerId;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AbstractTitusEnableDisableAction {
+  private final Logger log = LoggerFactory.getLogger(getClass());
+
+  private final AccountCredentialsProvider accountCredentialsProvider;
+  private final TitusEurekaSupport discoverySupport;
+  private final TitusClientProvider titusClientProvider;
+
+  public AbstractTitusEnableDisableAction(
+      AccountCredentialsProvider accountCredentialsProvider,
+      TitusEurekaSupport discoverySupport,
+      TitusClientProvider titusClientProvider) {
+    this.accountCredentialsProvider = accountCredentialsProvider;
+    this.discoverySupport = discoverySupport;
+    this.titusClientProvider = titusClientProvider;
+  }
+
+  /**
+   * Mark a Titus job and containing tasks as UP or OUT_OF_SERVICE in discovery _and_ any associated
+   * load balancers.
+   *
+   * @param saga Enclosing saga.
+   * @param description Enclosing description.
+   * @param shouldDisable Whether or not job and tasks should be marked OUT_OF_SERVICE (true) or UP
+   *     (false)
+   */
+  void markJob(Saga saga, EnableDisableServerGroupDescription description, boolean shouldDisable) {
+    String presentParticipling = shouldDisable ? "Disabling" : "Enabling";
+    String verb = shouldDisable ? "Disable" : "Enable";
+
+    saga.log(
+        "%s server group %s/%s",
+        presentParticipling, description.getRegion(), description.getServerGroupName());
+
+    NetflixTitusCredentials credentials =
+        (NetflixTitusCredentials)
+            accountCredentialsProvider.getCredentials(description.getAccount());
+    description.setCredentials(credentials);
+
+    try {
+      TitusClient titusClient =
+          titusClientProvider.getTitusClient(description.getCredentials(), description.getRegion());
+
+      TitusLoadBalancerClient titusLoadBalancerClient =
+          titusClientProvider.getTitusLoadBalancerClient(
+              description.getCredentials(), description.getRegion());
+
+      String serverGroupName = description.getServerGroupName();
+      String region = description.getRegion();
+
+      Job job = titusClient.findJobByName(serverGroupName, true);
+      if (job == null) {
+        saga.log("No job named '%s' found in %s", serverGroupName, region);
```

Comment:
should this sufficient for debugging or do we also need `log.error()` ?

Reply:
`saga.log()` does end up being logged to the standard logger.